src/python.cpp

```diff
@@ -3086,3 +3086,25 @@ SEXP py_has_method(PyObjectRef object, const std::string& name) {
 
   return Rf_ScalarLogical(result);
 }
+
+
+//' Unique identifer for Python object
+//'
+//' Get a globally unique identifier for a Python object.
+//'
+//' @note In the current implementation of CPython this is the
+//'  memory address of the object.
+//'
+//' @param object Python object
+//'
+//' @return Unique identifer (as string) or `NULL`
+//'
+//' @export
+// [[Rcpp::export]]
+SEXP py_id(PyObjectRef object) {
+  if (py_is_null_xptr(object))
+    return R_NilValue;
+
+  auto id((uintptr_t) object.get());
+  return CharacterVector({std::to_string(id)});
```

Comment:
Minor nit, but `std::to_string` requires C++11 -- do we need to declare that? Or, we could use a `std::stringstream`?

Reply:
Not minor at all, thanks for pointing this out! I did a quick search before writing this and noticed [`to_string()`](https://github.com/rstudio/reticulate/blob/d096cb782f6d8d1ceb8cfbf1b6c28b6cda2b3984/src/python.cpp#L95-L102), and at a cursory glance assumed it was fine to use... but on closer look maybe not. Perhaps `sprintf()`?

Reply:
Thus far we've supported the now-ancient C++98 standard, but now that R itself defaults to C++14 (IIRC) we could probably start using C++11 facilities in reticulate.

Reply:
I'll aim to keep the current compatibility this release; we can bump C++ std version the same time we drop python 2 support.